### PR TITLE
Fixes whitespace and run loading bugs in Muon GUIs

### LINKED ISSUE #32004: Muon Load bug
### Expected behavior
On any muon interface except the ALC interface Loading EMU00020882- 3 should load EMU00020882 and EMU00020883.

### Actual behavior
Loading EMU00020882- 3 raises an error. due to the space in between the hyphen and the last number and occurs with other runs as well.

### Steps to reproduce the behavior
1. Open Muon analysis interface .
2.  change instrument to EMU and in loading widget load 00020882- 3  and it should raise an error.
3. Error occurs for other instruments and runs as long as there is a space.

### Platforms affected
Confirmed on Windows but expected to be present on all platforms.

### PR DESCRIPTION
**Pull request by sf1919**
**Description of work.**
Changes to the run utils to deal with errant whitespaces in runs (e.g. 60715- 7) and runs that straddle a decade but the second number was lower than the final digit of the first  (e.g. 60719-2). The changes mean that neither issue produces an error message or fails to load at all.

**To test:**
1. Open Muon Analysis 2 GUI
2. Make sure the instrument selected is MUSR
3. Load runs ```62259-2``` 
4. An error message will appear about field alignments but the runs will still load (previously nothing would load at all)
5. Change instrument to EMU
6. Load runs 20882- 3
7. Runs should load correctly
8. Open Frequency Domain Analysis GUI and repeat steps 2 to 7

Fixes #32004 and #32073

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Review by tolu28-coder — DISMISSED** [review]
Everything works as expected

**Comment by Pasarus** [comment]
I think the solution here is a little ambiguous (for the multiple run loading bug). As far as I see it there are 3 solutions.
With an example of `40016-2` as a run number. 
1. The one that was implemented: [40016, 40017, ... 40023]
2. Decrement from: [40016, 40015, 40014, 40013, 40012]
3. Error out and suggest entering `40016-23` by popping up a window.

**Comment by AnthonyLim23** [comment]
I think users in general will put 40016-23 and that why users have not said anything. However, they do always use ascending order

**Comment by Pasarus** [comment]
Because of it's ambiguity perhaps 3. would be the better solution?

**Comment by AnthonyLim23** [comment]
honestly, I dont think a user would ever expect anything other than for it to increment.

**Comment by Pasarus** [comment]
It seems very ambiguous to me I don't think that it's useful to leave ambiguous behavior in our code base, error and suggesting a correct input would be better, as scientists/users will only ever do this by mistake.

**Comment by AnthonyLim23** [comment]
The muon users only enter runs in acceding order. They are more likely to load each run individually than to use the multiple load.

**Comment by Pasarus** [comment]
There is no case in which a user will expect this behavior, it's ambiguous in how it will behave, how do we know they didn't intend to get the next 10 or the next 10 after that as well? We don't and that's because it's ambiguous and because a user made a mistake in putting this in, in the first place. A warning that will suggest what they may have meant would be better, in my opinion, do you not agree?

I agree that most people will expect an ascending order, and I concede that we shouldn't necessarily support descending order, but because of this ambiguity from a UX perspective it makes sense to confirm what the user actually wants to do, and ask for exact input and not just guess and maybe get it wrong.

**Comment by AnthonyLim23** [comment]
I dont think this is an issue that will often come up. Also most users would specify more carefully if they did use a range (most dont), such as 2345-51

File: docs/source/release/v6.2.0/muon.rst [hunk 1]
```diff
@@ -18,13 +18,13 @@ Improvements
 
 - When running the Dynamic Kubo Toyabe fit function you should now be able to see the BinWidth to 3 decimal places.
 
-BugFixes
+Bug Fixes
 ############
```
Comment (gvardany) [inline docs/source/release/v6.2.0/muon.rst:22]: Can you fix the #s to match the title length?

File: docs/source/release/v6.2.0/muon.rst [hunk 2]
```diff
@@ -40,6 +40,13 @@ Improvements
 - Added a table to store phasequads in the phase tab, phasequads also no longer automatically delete themselves
   when new data is loaded
 
+Bug Fixes
+############
```
Comment (gvardany) [inline docs/source/release/v6.2.0/muon.rst:44]: Same here.

**Review by AnthonyLim23 — APPROVED** [review]

**Pull request merged**